src/measurements/conformance/ConformanceCriteria.js

```diff
@@ -1,162 +1,169 @@
+    return nonconformities
   }
 
   groupNonConformities(nonconformities) {
-    const groups = {};
-    const toolsGroupsMap = this.measurementApi.toolsGroupsMap;
+    const groups = {}
+    const toolsGroupsMap = this.measurementApi.toolsGroupsMap
 
     nonconformities.forEach(nonConformity => {
       if (nonConformity.isGlobal) {
-        groups.globals = groups.globals || { messages: [] };
-        groups.globals.messages.push(nonConformity.message);
+        groups.globals = groups.globals || { messages: [] }
+        groups.globals.messages.push(nonConformity.message)
 
-        return;
+        return
       }
 
       nonConformity.measurements.forEach(measurement => {
-        const groupName = toolsGroupsMap[measurement.toolType];
-        groups[groupName] = groups[groupName] || { measurementNumbers: {} };
+        const groupName = toolsGroupsMap[measurement.toolType]
+        groups[groupName] = groups[groupName] || { measurementNumbers: {} }
 
-        const group = groups[groupName];
-        const measureNumber = measurement.measurementNumber;
-        let measurementNumbers = group.measurementNumbers[measureNumber];
+        const group = groups[groupName]
+        const measureNumber = measurement.measurementNumber
+        let measurementNumbers = group.measurementNumbers[measureNumber]
 
         if (!measurementNumbers) {
           measurementNumbers = group.measurementNumbers[measureNumber] = {
             messages: [],
-            measurements: []
-          };
+            measurements: [],
+          }
         }
 
-        measurementNumbers.messages.push(nonConformity.message);
-        measurementNumbers.measurements.push(measurement);
-      });
-    });
+        measurementNumbers.messages.push(nonConformity.message)
+        measurementNumbers.measurements.push(measurement)
+      })
+    })
 
-    return groups;
+    return groups
   }
 
   validateTimepoint(timepointId, trialCriteriaType, data) {
-    const evaluators = this.getEvaluators(timepointId, trialCriteriaType);
-    let nonconformities = [];
+    const evaluators = this.getEvaluators(timepointId, trialCriteriaType)
+    let nonconformities = []
 
     evaluators.forEach(evaluator => {
-      const maxTargets = evaluator.getMaxTargets(false);
-      const maxNewTargets = evaluator.getMaxTargets(true);
+      const maxTargets = evaluator.getMaxTargets(false)
+      const maxNewTargets = evaluator.getMaxTargets(true)
       if (maxTargets) {
-        this.maxTargets = maxTargets;
+        this.maxTargets = maxTargets
       }
 
       if (maxNewTargets) {
-        this.maxNewTargets = maxNewTargets;
+        this.maxNewTargets = maxNewTargets
       }
 
-      const result = evaluator.evaluate(data);
-      nonconformities = nonconformities.concat(result);
-    });
+      const result = evaluator.evaluate(data)
+
+      if (result.length > 0) {
+        result.forEach(resultItem => {
+          resultItem.timepointType = timepointId
```

Comment:
Could you refactor `timepointId` to `timepointType` here and in its related callers as well?

Reply:
done. :)